### PR TITLE
Feature/dsnd 611 retry mechanism

### PR DESCRIPTION
**Pull request by vtondepu**

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/company/metrics/exception/RetryableTopicErrorInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+public class RetryableTopicErrorInterceptor implements ProducerInterceptor<String, Object> {
+
+    @Override
+    public ProducerRecord<String, Object> onSend(ProducerRecord<String, Object> record) {
+        String nextTopic = record.topic().contains("-error") ? getNextErrorTopic(record)
+                : record.topic();
+        LoggingConfig.getLogger().info(format("Moving record into new topic: %s with value: %s",
+                nextTopic, record.value()));
+        if (nextTopic.contains("-invalid")) {
```
Comment (zahidm3) [inline src/main/java/uk/gov/companieshouse/company/metrics/exception/RetryableTopicErrorInterceptor.java:23]: Can you please integrate Logger and log the topic its forwarding it, have a look at charges delta consumer for reference

File: src/main/java/uk/gov/companieshouse/company/metrics/processor/CompanyMetricsProcessor.java [hunk 2]
```diff
@@ -96,5 +97,26 @@ private String extractCompanyNumber(String resourceUri) {
...
+            final String logContext,
+            final Map<String, Object> logMap)
+            throws NonRetryableErrorException, RetryableErrorException {
+        logMap.put("status", httpStatus.toString());
+        if (HttpStatus.BAD_REQUEST == httpStatus) {
+            // 400 BAD REQUEST status is not retry-able
+            String message = "400 BAD_REQUEST response received from company-metrics-api";
+            logger.errorContext(logContext, message, null, logMap);
+            throw new NonRetryableErrorException(message);
+        } else if (httpStatus.is4xxClientError() || httpStatus.is5xxServerError()) {
```
Comment (zahidm3) [inline src/main/java/uk/gov/companieshouse/company/metrics/processor/CompanyMetricsProcessor.java:111]: Can you please check these if conditions in charges delta consumer

Comment (vtondepu) [reply src/main/java/uk/gov/companieshouse/company/metrics/processor/CompanyMetricsProcessor.java:111]: in Charges delta consumer i could see - !httpStatus.is2xxSuccessful(). Do you want me to change it to be inline with charges consumer? I am thinking if any 3xx or 1xx status codes they all fall under the retry.

File: src/main/java/uk/gov/companieshouse/company/metrics/exception/RetryableTopicErrorInterceptor.java (hunk 1 above)
Comment (vtondepu) [reply src/main/java/uk/gov/companieshouse/company/metrics/exception/RetryableTopicErrorInterceptor.java:23]: This has already done and is inline with charges delta consumer.

File: src/main/java/uk/gov/companieshouse/company/metrics/processor/CompanyMetricsProcessor.java (hunk 2 above)
Comment (vtondepu) [reply src/main/java/uk/gov/companieshouse/company/metrics/processor/CompanyMetricsProcessor.java]: As discussed updated the condition.

**Pull request merged**